### PR TITLE
Initialize DiscoveryClient

### LINKED ISSUE #517: KubernetesDiscoveryClient not registered with DiscoveryClientHealthIndicator
**Describe the bug**
I've encountered an interesting issue where the KubernetesDiscoveryClient is being instantiated and working for Kubernetes service discovery, yet it is never registered with the DiscoveryClientHealthIndicator.

The chain of events are as expected: https://cloud.spring.io/spring-cloud-static/Greenwich.SR4/single/spring-cloud.html#_health_indicator
- The DiscoveryClientHealthIndicator is created with a DependencyObjectProvider (Order.HIGHEST_PRECIDENCE)
- The KubernetesDiscoveryClient is created.
- The CompositeDiscoveryClient is created with instances of the SimpleDiscoveryClient and the KubernetesDiscoveryClient.

However, the DiscoveryClientHealthIndicator's onApplicationEvent callback is never called.  On inspection, I can see the ObjectProvider discoveryClient correctly references the CompositeDiscoveryClient.

The end result is the health indicator for the client is reported as "UNKNOWN".

Digging further, I see there are only two places where the expected InstanceRegisteredEvent could be called, but neither of these seem to be invoked:

AbstractAutoServiceRegistration
KubernetesAutoServiceRegistration (marked as deprecated)

Observed on: Spring Cloud Greenwich.SR4

### PR DESCRIPTION
**Pull request by ryanjbaxter**
Fixes #517

### CODE REVIEW

**Comment by ryanjbaxter** [comment]
Since all of these files look much different in 1.0.x I need to manually backport this.

**Comment by ryanjbaxter** [comment]
@spencergibb updated based on feedback

File: spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClientAutoConfiguration.java [hunk 1]
```diff
@@ -96,17 +93,13 @@ public KubernetesDiscoveryProperties getKubernetesDiscoveryProperties() {
...
-
-		@PostConstruct
-		public void postConstruct() {
-			this.indicatorInitializer.initialize();
+		@Bean
+		public KubernetesDiscoveryClientHealthIndicatorInitializer indicatorInitializer(
+				ApplicationEventPublisher applicationEventPublisher, PodUtils podUtils) {
+			KubernetesDiscoveryClientHealthIndicatorInitializer indicatorInitializer = new KubernetesDiscoveryClientHealthIndicatorInitializer(
+					podUtils, applicationEventPublisher);
+			indicatorInitializer.initialize();
```
Comment (spencergibb) [inline spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClientAutoConfiguration.java:101]: If your bean implements `IntializingBean` you wouldn't need to call this explicitly.

**Comment by spencergibb** [comment]
Looks like there are a bunch of unrelated commits?

File: spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClientAutoConfiguration.java (hunk 1 above)
Comment (ryanjbaxter) [reply spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClientAutoConfiguration.java]: I didnt know about that, updated PR!

**Comment by ryanjbaxter** [comment]
I will squash the commits before I merge it.

**Review by spencergibb — APPROVED** [review]

**Pull request merged**